third_party/nvfuser/csrc/scheduler/matmul.cpp

```diff
@@ -325,4 +334,310 @@ void scheduleMatmul(
   }
 }
 
+void scheduleMatmul(Fusion* fusion, const MatmulParams& params) {
+  const auto& inputs = fusion->inputs();
+  const auto& outputs = fusion->outputs();
+
+  TORCH_CHECK(
+      inputs.size() == 2,
+      "scheduleMatmul supports only fusions with two inputs");
+  TORCH_CHECK(
+      outputs.size() == 1,
+      "scheduleMatmul supports only fusions with single output");
+
+  TORCH_CHECK(
+      inputs[0]->isA<TensorView>(),
+      "fusion's first inpus is not an instance of TensorView class");
+  TORCH_CHECK(
+      inputs[1]->isA<TensorView>(),
+      "fusion's second inpus is not an instance of TensorView class");
+  TORCH_CHECK(
+      outputs[0]->isA<TensorView>(),
+      "fusion's output is not an instance of TensorView class");
+
+  TensorView* a = dynamic_cast<TensorView*>(inputs[0]);
+  TensorView* b = dynamic_cast<TensorView*>(inputs[1]);
+  TensorView* c = dynamic_cast<TensorView*>(outputs[0]);
+
+  auto mma_builder =
+      MmaBuilder(params.mma_op, params.tile_sizes).layout(params.layout);
+  const auto& gemm_tile = params.tile_sizes;
+
+  // Including current tensor naming convention for reference,
+  //  this is very temporary and will change over time and
+  //  in fact the whole body of this function will
+  //  eventually be a set of utility functions for different
+  //  sections of matmul(fusion) kernels, with
+  //  each having its own build out to do.
+  //
+  // Current naming convention:
+  //
+  //  operands assumed in global memory : a, b
+  //
+  //  registers staging global load : ar, br (short for a/b read)
+  //
+  //  shared mem cache of operands : acw_smem, bcw_smem (short for a/b
+  //  cache_write smem)
+  //
+  //  registers at shared memory load output : acr, bcr (short for a/b cache
+  //  read)
+  //
+  //  register tensor input to the actual mma op: ab, bb (short for a/b
+  //  broadcasted)
+  //
+  //  accumulator register: cc (short for c cache)
+  //
+  //  result in global memory: c
+
+  // Currently only support a, b, c as fusion inputs/outputs
+  //  aka. no prolog and epilog fusion yet.
+  TORCH_CHECK(
+      c->isFusionOutput() && a->isFusionInput() && b->isFusionInput(),
+      "not supporting matmul fusion yet");
+  TORCH_CHECK(c->definition() && c->definition()->isA<MmaOp>());
+
+  mma_builder.configureMma(c);
+
+  // TODO:
+  // Beyond this point, mma_builder really just becomes a populated
+  //  list of parameters to describes the mma swizzles that should
```

Comment:
nit: describe

Reply:
Fixed in the updated sources, that are present in PR in new project ([link](https://github.com/NVIDIA/Fuser/pull/23))